cmd/query/app/http_handler.go

```diff
@@ -450,6 +451,11 @@ func (aH *APIHandler) writeJSON(w http.ResponseWriter, r *http.Request, response
 		}
 	}
 	resp, _ := marshall(response)
-	w.Header().Set("Content-Type", "application/json")
+
```

Comment:
Q: should this really be scoped only to the API endpoints? I thought it would be a top-level middleware for all of query-service (that's how we did security headers internally).

Reply:
So I'm not a CORs expert, but my understanding is that it's mainly used on APIs to give browser information about how the API should be accessible to a script in a webpage.  So, as I was working on this feature I was mainly thinking in terms of the API calls.

I'm a little concerned that applying the same headers to the API calls as the main page would have unintended side effects.  I'm not an expert on the behavior of these headers, but I'd be glad to add that functionality.

Reply:
@everett980 @EranOfer do you have thoughts on this?

Reply:
If you'd like I can just go ahead and headers to all jaeger-query endpoints.  If it's working internally at Jaeger then it's probably fine.

Reply:
Yes, let's do it as a top-level middleware on all of query-service HTTP endpoints. If someone has specific needs later on to restrict to some endpoints only, we can revisit it later.